internal/confmapprovider/discovery/config.go

```diff
@@ -264,7 +315,10 @@ func loadEntry[K keyType, V entryType](componentType, path string, target map[K]
 	tmpDest := map[K]V{}
 
 	componentID, err := unmarshalEntry(componentType, path, &tmpDest)
-	noTypeK := componentIDToK(noType, componentID)
+	noTypeK, err2 := stringToKeyType(noType.String(), componentID)
```

Comment:
`K` and `V` are type parameters whose type constraints are `keyType` and `entryType`. The `K` type argument (in the function body at run time) is the actual type used in the generic function invocation and we need to cast the `""` provided by `noType.String()` to this type in order to get the zero value of type `V` in `tmpDest[noTypeK]` to call `ErrorF()`. If `stringToKeyType()` fails (errors) then presumably the value of `noTypeK` shouldn't be used so we'd error out early there. In practice this isn't really the case since the only error path would be for a breaking change in `config.NewComponentIDFromString()`.